breadbox/breadbox/models/dataset.py

```diff
@@ -116,6 +116,7 @@ class TabularDataset(Dataset):
     index_type_name = Column(
         String, ForeignKey("dimension_type.name", ondelete="CASCADE"), nullable=False
     )
+    index_type = relationship("DimensionType", foreign_keys=[index_type_name])
```

Comment:
I added this relationship field so it matches the `sample_type` and `feature_type` fields we have on the matrix datasets.